LibraBFT/Impl/Consensus/RoundManager.agda

```diff
@@ -56,25 +57,28 @@ module processProposalMsgM (now : Instant) (pm : ProposalMsg) where
   step₂ : Either FakeErr Bool → LBFT Unit
 
   step₀ =
-    case pm ^∙ pmProposer of λ where
+    caseMM pm ^∙ pmProposer of λ where
       nothing → logInfo -- log: info: proposal with no author
       (just pAuthor) → step₁ pAuthor
   step₁ pAuthor =
         ensureRoundAndSyncUpM now (pm ^∙ pmProposal ∙ bRound) (pm ^∙ pmSyncInfo)
                               pAuthor true >>= step₂
-  step₂ r =
-        -- IMPL-DIFF: We use `ifM` to test whether the round of the proposal is
-        -- current, to take advantage of the obligations `RWST-weakestPre` generates.
-        case r of λ where
-        (Left _) → logErr -- log: error: <propagate error>
-        (Right pmCurrent) →
-          if pmCurrent
-            then processProposalM (pm ^∙ pmProposal)
-            else do
-              currentRound ← use (lRoundState ∙ rsCurrentRound)
-              logInfo  -- log: info: dropping proposal for old round
-
-processProposalMsgM = processProposalMsgM.step₀
+  step₂ = λ where
+          -- IMPL-DIFF: We use `ifM` to test whether the round of the proposal is
+          -- current, to take advantage of the obligations `RWST-weakestPre` generates.
+          (Left _) → logErr -- log: error: <propagate error>
```

Comment:
`fakeErr` does not have the same type as `logErr`.  The places where we use `fakeErr` are where the Haskell code returns an error, and `logErr` are where the Haskell code outputs and error.

I'm not sure what motivates the different treatment in the Haskell code. If there is a good reason for the prototype to more consistently produce errors as output, rather than return them, then I'm ok with the change -- but any such change would require re-doing the proofs, so if there *isn't* a compelling reason then I would leave it as is

Reply:
Oh yes, return versus log.  Got it.
I agree, no change necessary.
Haskell returns errors from pure functions (and some LBFT ones too, to follow the Rust code).
And Haskell always logs errors from "top-level" LBFT functions (most of the ones in RoundManager).